lib/ansible/module_utils/facts/system/pkg_mgr.py

```diff
@@ -113,12 +113,22 @@ def _check_apt_flavor(self, pkg_mgr_name):
                     pkg_mgr_name = 'apt'
         return pkg_mgr_name
 
+    def pkg_mgrs(self, collected_facts):
+        # Filter out the /usr/bin/pkg because on Altlinux it is actually the
+        # perl-Package (not Solaris package manager).
+        # Since the pkg5 takes precedence over apt, this workaround
```

Comment:
this is incorrect, apt-get has higher precedence than pkg

Reply:
It's correct:
https://github.com/ansible/ansible/blob/7830e5308e85707b5527f0769836ccfb3c8b1fb3/lib/ansible/module_utils/facts/system/pkg_mgr.py#L13-L15

Each next  ```pkg['name']``` in ```PKG_MGRS``` can rewrite the ```pkg_mgr_name```:
https://github.com/ansible/ansible/blob/7830e5308e85707b5527f0769836ccfb3c8b1fb3/lib/ansible/module_utils/facts/system/pkg_mgr.py#L121-L123

Reply:
You are correct, I didn't look at all the code and assumed assumed there was a break on match, its last, not first, found wins